onnxruntime_extensions/pnp/_torchext.py

```diff
@@ -268,4 +268,35 @@ def export(self, *args, **kwargs):
             return super().export(*args, **kwargs)
         else:
             oxml = core_m.export(*args, **kwargs)
-            return ONNXModelUtils.join_models(prefix_m, oxml)
+            model = ONNXModelUtils.join_models(prefix_m, oxml)
+
+            # Rename the input/output node names if the user has provided any substitutions!
+            # Ref: https://github.com/onnx/onnx/issues/2052
+            if (('input_names' in kwargs) or ('output_names' in kwargs)) and \
+              (kwargs['input_names'] or kwargs['output_names']):
+                swaps = {}
+                if 'input_names' in kwargs and kwargs['input_names']:
+                    assert len(model.graph.input) == len(kwargs['input_names']), \
+                      "Expecting {} input names but got {}".format(
+                        len(model.graph.input), len(kwargs['input_names']))
+                    for n, new_name in zip(model.graph.input, kwargs['input_names']):
+                        swaps[n.name] = new_name
+                        n.name = new_name
+
+                if 'output_names' in kwargs and kwargs['output_names']:
+                    assert len(model.graph.output) == len(kwargs['output_names']), \
+                      "Expecting {} output names but got {}".format(
+                        len(model.graph.output), len(kwargs['output_names']))
+                    for n, new_name in zip(model.graph.output, kwargs['output_names']):
+                        swaps[n.name] = new_name
+                        n.name = new_name
+
+                if swaps:
+                    for n in model.graph.node:
+                        for j in range(len(n.input)):
```

Comment:
also need to update the model.graph.initialize, which also contains the name.

Reply:
Done!